protocols/kad/src/behaviour.rs

```diff
@@ -268,10 +300,27 @@ impl<TSubstream> Kademlia<TSubstream> {
     }
 
     /// Returns an iterator to all the peer IDs in the bucket, without the pending nodes.
-    pub fn kbuckets_entries(&self) -> impl Iterator<Item = &PeerId> {
-        self.kbuckets.entries_not_pending().map(|(kad_hash, _)| kad_hash.peer_id())
+    pub fn kbuckets_entries_not_pending<'a>(&'a self) -> impl 'a + Iterator<Item = &PeerId> {
```

Comment:
I don't think we need both a version with pending nodes and non-pending nodes.
Pending nodes aren't part of the k-buckets in my opinion, so we should either have a method that considers them not-part-of-buckets, or rename `kbuckets_entries_not_pending` to `kbuckets_entries` and `kbuckets_entries` to `kbuckets_entries_pending`.

Reply:
The renaming scheme sounds easiest to me right now.

Reply:
Well, option #1 consists in removing a method.